mapper/tedge_mapper/src/main.rs

```diff
@@ -121,7 +121,9 @@ fn check_another_instance_is_running(app_name: &str) -> Result<Flockfile, Flockf
 }
 
 fn config_repository() -> Result<TEdgeConfigRepository, MapperError> {
-    let tedge_config_location = if tedge_users::UserManager::running_as_root() {
+    let tedge_config_location = if tedge_users::UserManager::running_as_root()
```

Comment:
This API is used by other mappers like `collectd_mapper` as well, for which the user is `collectd-mapper`. Then this logic would be wrong. So, similar to the suggestion above, take the `desired_user` as a parameter in this API as well and then do the validation.

Reply:
I don't fully agree here. `collectd_mapper` doesn't access to tedge config. This is a function in `tedge_mapper` crate. Hard-code `tedge-mapper` user seems not a problem for me.
However, it depends on what we will decide. If we divide `tedge-mapper` user into `tedge-mapper-c8y` and `tedge-mapper-az` users, then I agree with your suggestion.

Reply:
The `main.rs` of `collectd-mapper` uses this API in the `mqtt_port()` method. And with the current code, even collectd will be just checking if the user is `tedge-mapper` instead of `collectd-mapper` and fall back to the config file in the home directory.

Reply:
Yeah, I added the `main.rs` of `collectd-mapper`. But collectd-mapper always worked by another bug :/

The story is:
1. When collectd-mapper is running as collectd-mapper user, the mapper always connects on port 1883 because of this bug.
2. However, mosquitto opens the port 1883 always because of `/etc/tedge/mosquitto-conf/tedge-mosquitto.conf` has `lister 1883`.

So, it was very hard to find this bug :/